### ISSUE TITLE
A single instance of disabling DeclarationOrder is preventing the CLI from finishing

### ISSUE THREAD

**Issue by ackernaut** [causes/scss-lint#378]
I have a project with multiple Sass partials, organized in directories and sub-directories—typical. For some reason version `0.34.0` would not finish linting from console. I traced it down to a single file, and a single instance of `// scss-lint:disable DeclarationOrder`.

```scss
.navbar-user-dropdown-toggle {
  // scss-lint:disable DeclarationOrder
  @extend .clearfix;
  @extend %offcanvas-push;

  display: block;
  padding: 0 .75rem;

  // &:hover {
  //   background-color: $navbar-hover-transparency;
  // }

  .open > &,
  .open > &:hover {
    background-color: transparent;
  }

  @include bp($qfk-screen-sm-min) {
    padding: 0 1rem;
  }

  // Angle arrow icon
  // ----------------
  // Note: See responsive utility class usage in markup

  > .fa {
    @include navbar-height;

    color: $navbar-mid-grey;
    display: block;
    float: left;
    font-size: 14px;
    margin-left: .5rem;

    // When the navbar is transparent
    // ------------------------------

    .navbar-is-transparent & {
      color: $navbar-transparent-link-color;
    }
  }

  > .fa-angle-left,
  > .fa-angle-right,
  > .fa-angle-up,
  > .fa-angle-down {
    display: none;
  }

  @include bp($max-width: $qfk-screen-sm-max) {
    > .fa-angle-left {
      display: none;

      .has-open-navbar-user-menu & {
        display: block;
      }
    }

    > .fa-angle-right {
      display: block;

      .has-open-navbar-user-menu & {
        display: none;
      }
    }
  }

  @include bp($qfk-screen-md-min) {
    > .fa-angle-down {
      display: block;
    }
  }

  // When the navbar is transparent
  // ------------------------------

  .navbar-is-transparent & {
    @include link-me($color: $navbar-transparent-link-color, $hover-color: $navbar-transparent-link-hover-color);
  }
}
```

If I remove that line everything works fine. There are other instances in the same file of the same disabling comment (same linter, first comment inside of a selector block with nested blocks).

If I wrap the selector it works fine.

```scss
// scss-lint:disable DeclarationOrder
.navbar-user-dropdown-toggle {
  // ...
}
// scss-lint:enable DeclarationOrder
```

I thought it strange that it would break the linting task entirely. I know that is a messy block of code, but was it an error on my part?

**Comment by sds** [comment]
Nope, there is no situation where `scss-lint` looping forever would be due to user error. :)

I've fixed this in 0cbacccdaab3e0bb7e375626774421c68bec9f89. Thanks for the report!

**Comment by ackernaut** [comment]
:+1: thank you!
